addons/mail/static/src/models/thread/thread.js

```diff
@@ -724,18 +724,17 @@ function factory(dependencies) {
          * Unsubscribe current user from provided channel.
          */
         async unsubscribe() {
-            if (this.channel_type === 'mail.channel') {
-                return this.async(() => this.env.services.rpc({
+            const wasPinned = this.pinnedStatus;
+            this.update({ pinnedStatus: 'pending_unpin' });
+            try {
```

Comment:
What's the motivation for `try`/`catch` here?
(If it's to force an `.update()` on deleted thread, this is wrong)

Reply:
it was not because of delete()
The try and catch is here to be able to revert the change if the rpc fails
I need to update the thread before the rpc to have immediate effect on the interface, and start the pending_unpin > unpin transition process.

I changed it not to incluse this.async in the try...catch

Reply:
Ok, then indeed it's good `try`/`catch` just around RPC